File: backend/Models/Product.cs
namespace backend.Models
{
    public class Product : BaseModel
    {
        public string Title { get; set; } = null!;
        public double Price { get; set; }
        public string? Description { get; set; }
        public Category Category { get; set; } = null!;
        public ICollection<Image>? Images { get; set; }

    }
}

File: frontend/src/types/product.ts
import { Category } from "./category"
import { Image } from "./Image"

export interface Product {
    id: number
    title: string
    description:string
    price:number
    category: Category
    images:Image[]
    quantity?:number
    subTotal?:number
}


export interface ModifyProduct{
    updateId:number|undefined
    update:{
        title?:string
        description?:string
        price?:number
        category?: string
    }
    
}

export interface Iprops {
    title?:string
    price?: number
    description?: string
    category?: string
    images?: string[]
    create?: string
    update?: string
    cancel?: string 
}